src/aks-preview/azext_aks_preview/_help.py

```diff
@@ -198,6 +198,13 @@
           type: string
           short-summary: The Kubernetes network plugin to use.
           long-summary: Specify "azure" for routable pod IPs from VNET, "kubenet" for non-routable pod IPs with an overlay network, or "none" for no networking configured.
+        - name: --network-plugin-mode
+          type: string
+          short-summary: The network plugin mode to use.
+          long-summary: |
+              Specify "overlay" to use an overlay network built on Azure networking.
+              Differs from Kubenet overlay as it does not require route tables.
+              Must be used with --network-plugin=azure.
```

Comment:
Could you add a validator for this point?

Reply:
Updated the description. Long term, the plugin mode will be able to be used with other network plugins and we should allow aks-rp to handle the validation instead.